cpp/src/lists/contains.cu

```diff
@@ -0,0 +1,266 @@
+                                    counting_iter(list.size()),
+                                    [&list] __device__(auto const& i) { return list.is_null(i); });
+                                });
+}
+
+struct lookup_functor {
+  template <typename T, typename... Args>
+  std::enable_if_t<!cudf::is_numeric<T>() && !std::is_same<T, cudf::string_view>::value, void>
+  operator()(Args&&...) const
+  {
+    CUDF_FAIL("lists::contains() is only supported on numeric types and strings.");
+  }
+
+  template <typename T>
+  std::enable_if_t<cudf::is_numeric<T>() || std::is_same<T, cudf::string_view>::value, void>
+  operator()(cudf::detail::lists_column_device_view const& d_lists,
+             cudf::scalar const& skey,
+             cudf::mutable_column_device_view output_bools,
+             rmm::cuda_stream_view stream,
+             rmm::mr::device_memory_resource* mr) const
+  {
+    assert(skey.is_valid() && "skey should have been checked for nulls by this point.");
+
+    auto h_scalar = static_cast<cudf::scalar_type_t<T> const&>(skey);
+    auto d_scalar = cudf::get_scalar_device_view(h_scalar);
+
+    thrust::transform(rmm::exec_policy(stream),
+                      counting_iter(0),
+                      counting_iter(d_lists.size()),
+                      output_bools.data<bool>(),
+                      [d_lists, d_scalar] __device__(auto row_index) {
+                        auto list = cudf::list_device_view(d_lists, row_index);
+                        if (list.is_null()) { return false; }
+                        for (size_type i{0}; i < list.size(); ++i) {
+                          if (list.is_null(i)) { return false; }
+                          auto list_element = list.template element<T>(i);
+                          if (list_element == d_scalar.value()) { return true; }
+                        }
+                        return false;
+                      });
+  }
+
+  template <typename T>
+  std::enable_if_t<cudf::is_numeric<T>() || std::is_same<T, cudf::string_view>::value, void>
+  operator()(cudf::detail::lists_column_device_view const& d_lists,
+             cudf::column_device_view const& d_skeys,
+             cudf::mutable_column_device_view output_bools,
+             rmm::cuda_stream_view stream,
+             rmm::mr::device_memory_resource* mr) const
+  {
+    thrust::transform(rmm::exec_policy(stream),
+                      counting_iter(0),
+                      counting_iter(d_lists.size()),
+                      output_bools.data<bool>(),
+                      [d_lists, d_skeys] __device__(auto row_index) {
+                        if (d_skeys.is_null(row_index)) { return false; }
+                        auto list = cudf::list_device_view(d_lists, row_index);
+                        if (list.is_null()) { return false; }
+                        auto skey = d_skeys.template element<T>(row_index);
+                        for (size_type i{0}; i < list.size(); ++i) {
```

Comment:
Same comment as above about using a `pair_iterator` and `thrust::find_if(thrust::seq, ...)`

Reply:
This is now handled in the common iterator-based helper function, via a `pair_iterator`.